File: usc_utils/blackboard/test/test_blackboard_client.py
#!/usr/bin/env python
import roslib
roslib.load_manifest('blackboard')
import rospy

from blackboard.blackboard_client import BlackBoardClient

rospy.init_node('BlackBoardClientTest', log_level = rospy.DEBUG)
rospy.loginfo('Started BlackBoardClientTest node.')

blackboard_client = BlackBoardClient()
blackboard_client.initialize('left')
rospy.sleep(0.5)
rospy.loginfo("Waiting for publisher to connect.")

while not rospy.is_shutdown():

    blackboard_client.debug('key1', 'value1')
    rospy.sleep(0.5)
    blackboard_client.info('key2', 'value1')
    rospy.sleep(0.5)
    blackboard_client.warn('key3', 'value1')
    rospy.sleep(0.5)
    blackboard_client.error('key4', 'value1')
    rospy.sleep(0.5)
    blackboard_client.fatal('key5', 'value1')
    rospy.sleep(0.5)
    blackboard_client.white('key1')
    rospy.sleep(0.2)
    blackboard_client.yellow('key2')
    rospy.sleep(0.2)
    blackboard_client.red('key3')
    rospy.sleep(0.2)
    blackboard_client.purple('key4')
    rospy.sleep(0.2)
    blackboard_client.green('key5')
    rospy.sleep(0.2)
    blackboard_client.prediction('whatever')
    rospy.sleep(0.2)

File: usc_utils/blackboard/test/test_blackboard_client.cpp
/*
 * test_blackboard_client.cpp
 *
 *  Created on: Oct 13, 2013
 *      Author: pastor
 */




// system includes
#include <ros/ros.h>
// local includes
#include <blackboard/blackboard_client.h>

int main(int argc, char** argv)
{
  ros::init(argc, argv, "TestBlackBoardClient");
  ros::NodeHandle node_handle("~");

  blackboard::BlackBoardClient right_blackboard_client;
  if(!right_blackboard_client.initialize("right"))
    return -1;
  blackboard::BlackBoardClient left_blackboard_client;
  if(!left_blackboard_client.initialize("left"))
    return -1;

  while (ros::ok())
  {
    right_blackboard_client.debug("key1", "value1");
    left_blackboard_client.debug("key1", "value1");

    ros::spinOnce();
    ros::Duration(1.0).sleep();

    right_blackboard_client.info("key2", "value2");
    left_blackboard_client.info("key2", "value2");

    ros::spinOnce();
    ros::Duration(1.0).sleep();

    right_blackboard_client.warn("key3", "value3");
    left_blackboard_client.warn("key3", "value3");

    ros::spinOnce();
    ros::Duration(1.0).sleep();

    right_blackboard_client.error("key4", "value4");
    left_blackboard_client.error("key4", "value4");

    ros::spinOnce();
    ros::Duration(1.0).sleep();

    right_blackboard_client.fatal("key5", "value5");
    left_blackboard_client.fatal("key5", "value5");
  }
  return 0;
}
